### PR TITLE
pkgs/by-name/[ab].+: migrate to new apple sdk

### PR DESCRIPTION
**Pull request by sarahec**
Validating the new apple sdk by migrationg low-risk packages:
- At most 1 downstream dependency
- Only replacing Darwin-specific BuildInputs
- No impact on other platforms

1. Replaced `darwin` dependency with `apple-sdk_11`
2. Replaced all references to darwin.apple_sdk.frameworks.* with single `apple-sdk_11`
3. Reformatted with current nixfmt (rfc style)
4. Minor fixes as needed such as dropping "with lib;"

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sarahec** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354097`

---
### `x86_64-linux`

File: pkgs/by-name/ag/agate/package.nix [hunk 1]
```diff
@@ -30,7 +30,7 @@ rustPlatform.buildRustPackage rec {
     [ openssl ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin [
       libiconv
-      darwin.apple_sdk.frameworks.Security
+      apple-sdk_11
```
Comment (emilazy) [inline pkgs/by-name/ag/agate/package.nix:33]: This does not seem correct: the original package didn’t override the SDK at all, and `apple-sdk_11` is not the default on `x86_64-darwin`. If you want the default SDK, you don’t need to specify anything; non‐overridden `darwin.apple_sdk.frameworks.*` inputs in `by-name` should simply be dropped entirely. `libiconv` too.

Most likely the OpenSSL dependency is not used on Darwin, so it should instead be `buildInputs = lib.optionals (!stdenv.hostPlatform.isDarwin) [ openssl ];`, but as that was a pre‐existing problem it’s not a blocker for this PR.

**Review by emilazy — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ag/agate/package.nix (hunk 1 above)
Comment (sarahec) [reply pkgs/by-name/ag/agate/package.nix]: Oh! I misunderstood.

Changed just this one, if it looks good I'll go do the rest.

Comment (emilazy) [reply pkgs/by-name/ag/agate/package.nix]: Looks good to me. Any time we’re conditionalizing `openssl` on not‐Darwin we’ll need to check it still builds (some things *do* want OpenSSL on Darwin, especially if they’re not already including the `Security` framework), but just removing the frameworks and `libiconv` should be the way to go for anything that doesn’t specifically override the version currently.

**Comment by sarahec** [comment]
@emilazy That should be everything

**Review by emilazy — APPROVED** [review]
Thanks, looks good now! I did a survey of OpenSSL dependencies:

* `agate` uses `tokio-rustls` with `ring` and therefore shouldn’t require OpenSSL on any platform AFAIK; can you drop the `buildInputs` entirely?

* `asfa` uses `ssh2` which requires OpenSSL, so that seems fine.

* `asm-lsp` uses `reqwest` which uses native TLS by default; OpenSSL dependency should be conditioned on `!stdenv.hostPlatform.isDarwin`.

I won’t block this PR on this if you don’t feel like doing it, but it’d be nice to get these cleaned up.

**Comment by sarahec** [comment]
> Thanks, looks good now! I did a survey of OpenSSL dependencies:
> 
> * `agate` uses `tokio-rustls` with `ring` and therefore shouldn’t require OpenSSL on any platform AFAIK; can you drop the `buildInputs` entirely?

If I do so, it fails looking for <openssl/opensslv.h>. So it needs the headers at least.

> * `asfa` uses `ssh2` which requires OpenSSL, so that seems fine.
> * `asm-lsp` uses `reqwest` which uses native TLS by default; OpenSSL dependency should be conditioned on `!stdenv.hostPlatform.isDarwin`.

Done.

**Review by emilazy — APPROVED** [review]
Result of `nixpkgs-review pr 354097` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

## `nixpkgs-review` result                                                      
                                                                                
Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).    
                                                                                
Command: `nixpkgs-review pr 354097`                                             
                                                                                
---                                                                             
### `aarch64-darwin`                                                            
<details>                                                                       
  <summary>:white_check_mark: 13 packages built:</summary>                      
  <ul>                                                                          
    <li>agate</li>                                                              
    <li>anchor</li>                                                             
    <li>asfa</li>                                                               
    <li>asfa.man</li>                                                           
    <li>asm-lsp</li>                                                            
    <li>atac</li>                                                               
    <li>attract-mode</li>                                                       
    <li>avbroot</li>                                                            
    <li>bacon</li>                                                              
    <li>biome</li>                                                              
    <li>buckle</li>                                                             
    <li>bulloak</li>                                                            
    <li>commafeed</li>                                                          
  </ul>                                                                         
</details>

Thank you!

**Pull request merged**